base/iostream.jl

```diff
@@ -225,8 +225,18 @@ function readuntil(s::IOStream, delim::UInt8)
     ccall(:jl_readuntil, Array{UInt8,1}, (Ptr{Void}, UInt8, UInt8), s.ios, delim, 0)
 end
 
-function readline(s::IOStream)
-    ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+function readline(s::IOStream, chomp::Bool=false)
+    chomp || return ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+
+    s = ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
+    i = endof(s)
```

Comment:
`endof` is costly since it needs to look for the start of the last Unicode char. Here we're working at the byte level, so use `sizeof(s)` instead. Then, you cannot index directly into `s` as you might be in the middle of a character: you need to access the underlying array of bytes directly (cf. point below).